### PR TITLE
Reworks (mostly) and renames all the storytellers and improves the code.

### PR DESCRIPTION
**Pull request by BurgerLUA**
## About The Pull Request

Adds a new storyteller tag called TAG_CHAOS which is assigned to chaotic events. So far, this is assigned to Blob, Meteors, Abductor, Rampant Brand Intelligence, Space Vines, Portal Storms, MALF, Nuke Ops, Space Dragon, Xenomorph Infestation, and Spider Infestation.

All existing storytellers have been renamed and tweaked to what they're supposed to do, if they were known to have issues (see below).

Changes how antag cap calculation works. Makes it so that antag count is limited by security's population as well as population count.

### Current Storytellers
The Gamer: A remake of the warrior. Has a focus on combat events and antagonist spawns, while limiting destructive events.

Default Andy: The default storyteller. Just a renamed guide.

The Bomb: A remake of the DemoMan. Practically unchanged from The DemoMan.

Extended: A remake of The Ghost. Practically unchanged from The Ghost.

The Fragile: A working (allegedly) version that existed before. It had a tendency to spawn all the antagonists because an antag divisor wasn't set.

The Chill: A remake of The Guide. Limits chaotic, combat, and destructive events.

The Clown: A remake of The Jester. Practically unchanged from The Jester.

## Why It's Good For The Game

Storyteller seems to be having issues right now where non-chaotic storytellers are turning into chaos. Part of the reason for this is that people have different expectations of storytellers than what it actually does, and the storyteller system doesn't really have consideration if an event is chaotic.

According to Odairu, another issue that needs to be fixed is how the antag cap isn't updated while antags are being spawned, so that probably needs to be fixed in another PR.

Even though some storytellers weren't changed, I renamed all the storytellers as well to make it describe what it actually does. Not everyone plays Rimworld.

## Proof Of Testing

Untested.

## Changelog

:cl: BurgerBB
balance: Reworks (mostly) and renames all the storytellers and improves the code.
/:cl:

### CODE REVIEW

**Comment by Odairu** [comment]
>According to Odairu, another issue that needs to be fixed is how the antag cap isn't updated while antags are being spawned, so that probably needs to be fixed in another PR.

for reference, this is for round start where story teller is spending points to make ROUNDSTART antags, since before the round actually starts and people have loaded in, regardless of how many antags story teller rolls for, there are no antags, so it can roll 3 times and on the third time it doesn't know it rolled 2 times before

**Review by projectkepler-ru — APPROVED** [review]

**Review by nevimer — APPROVED** [review]
I like it, but whats up with the file moving

**Comment by BurgerLUA** [comment]
> I like it, but whats up with the file moving

For organization purposes, each storyteller gets its own file.

File: modular_zubbers/code/modules/storyteller/gamemode.dm [hunk 1]
```diff
@@ -213,7 +213,9 @@ SUBSYSTEM_DEF(gamemode)
 /datum/controller/subsystem/gamemode/proc/get_antag_cap()
 	if(isnull(storyteller))
 		return 0
-	return round(get_correct_popcount() / storyteller.antag_divisor) + (sec_crew ? sec_crew : ANTAG_CAP_FLAT) // Population divide by storyeteller's divisor, add one antag per sec.
+	if(storyteller.antag_divisor == 0)
+		return 0
+	return CEILING( clamp(get_correct_popcount() / storyteller.antag_divisor + sec_crew,ANTAG_CAP_FLAT,sec_crew*1.5), 1 )
```
Comment (Majkl-J) [inline modular_zubbers/code/modules/storyteller/gamemode.dm:218]: I'm not sure whether to trust ceiling in this, we used it before and for some reason it clamped the whole thing to one even though it makes no sense for it to do so. Whatever, we'll see in testing

```suggestion
	return CEILING(clamp(get_correct_popcount() / storyteller.antag_divisor + sec_crew, ANTAG_CAP_FLAT, sec_crew*1.5), 1)
```

File: modular_zubbers/code/modules/storyteller/storytellers/tellers/storyteller_peaceful.dm [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+		EVENT_TRACK_MODERATE = 0.7,
+		EVENT_TRACK_MAJOR = 0.7,
+		EVENT_TRACK_ROLESET = 0.7,
+		EVENT_TRACK_OBJECTIVES = 1,
+		)
+	guarantees_roundstart_roleset = FALSE
+	tag_multipliers = list(
+		TAG_COMBAT = 0.3,
+		TAG_DESTRUCTIVE = 0.3,
+		TAG_CHAOTIC = 0.1
```
Comment (Majkl-J) [inline modular_zubbers/code/modules/storyteller/storytellers/tellers/storyteller_peaceful.dm:16]: To be honest chaotic could be knocked down to 0 on this one but up to you really

File: modular_zubbers/code/modules/storyteller/storytellers/tellers/storyteller_peaceful.dm [hunk 3]
```diff
@@ -0,0 +1,18 @@
+/datum/storyteller/peaceful
+	name = "The Chill"
+	desc = "The Chill will be light on events compared to other storytellers, especially so on ones involving combat, destruction, or chaos. Best for more chill rounds."
+	welcome_text = "If you vote for this storyteller on Ice Box, you have no originality."
```
Comment (Majkl-J) [inline modular_zubbers/code/modules/storyteller/storytellers/tellers/storyteller_peaceful.dm:4]: Okay this one is funny

**Review by Majkl-J — CHANGES_REQUESTED** [review]
I like this, literally only a formatting nitpick (Also sobbing because this will cause conflicts with teller 2.0 for which I had the same idea but oh well I should've been faster)

Anyhow, burger cooking moment

**Review by Majkl-J — APPROVED** [review]
Has been working well, so I'll be looking to merge this whenever you consider it ready

**Comment by BurgerLUA** [comment]
todo: some events don't exist in the storyteller folder, and need to be added to that

**Comment by Swiftfeather** [comment]
Where's predictable chaos?

**Comment by BurgerLUA** [comment]
> Where's predictable chaos?

It should still exist in here

**Pull request merged**